### PR TITLE
feat: Linkup intergrate

### LINKED ISSUE #1345: [Feature Request] Integrate Linkup
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

https://docs.linkup.so/pages/get-started/quickstart

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by raywhoelse**
## Description

Describe your changes in detail.

## Motivation and Context

 `close #1345` 

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/toolkits/search_toolkit.py [hunk 1]
```diff
@@ -63,6 +64,156 @@ def search_wiki(self, entity: str) -> str:
...
+        query: str,
+        depth: Literal["standard", "deep"] = "standard",
+        output_type: Literal[
+            "searchResults", "sourcedAnswer", "structured"
+        ] = "searchResults",
+        structured_output_schema: Union[Type[BaseModel], str, None] = None,
+    ) -> Dict[str, Any]:
+        """
+        Search for a query in the Linkup API and return results in various
+        formats.
```
Comment (Wendong-Fan) [inline camel/toolkits/search_toolkit.py:80]: docstring format
```suggestion
        r"""Search for a query in the Linkup API and return results in various
        formats.
```

File: camel/toolkits/search_toolkit.py [hunk 2]
```diff
@@ -63,6 +64,156 @@ def search_wiki(self, entity: str) -> str:
...
+                - "sourcedAnswer" for an answer with supporting sources,
+                - "structured" for output based on a provided schema.
+            structured_output_schema (Union[Type[BaseModel], str, None]): If
+                `output_type` is "structured",specify the schema of the
+                output. Can be a Pydantic BaseModel or a JSON schema string.
+
+        Returns:
+            Dict[str, Any]: A dictionary representing the search result. The
+            structure depends on the `output_type`. If an error occurs,
+            returns an error message.
```
Comment (Wendong-Fan) [inline camel/toolkits/search_toolkit.py:99]: docstring format
```suggestion
        Returns:
            Dict[str, Any]: A dictionary representing the search result. The 
                structure depends on the `output_type`. If an error occurs, 
                returns an error message.
```

**Review by Wendong-Fan — COMMENTED** [review]
thanks @raywhoelse ! There are some minor issues with the docstring format that need to be fixed. For this search function, I feel it's not necessary to add four additional helper functions. While making the code more modular has its benefits, in this case, it adds redundancy without enough value to justify it. WDYT?

**Review by liuxukun2000 — APPROVED** [review]

**Review by Wendong-Fan — COMMENTED** [review]
Hi @raywhoelse , Please remember to run poetry.lock if you've updated the pyproject.toml file, also add unit test

Thanks @liuxukun2000's review, please kindly follow our [Code Review Checklist](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md#code-review-checklist) when approving PRs, the check for this PR is failing. Code review approvals should be even stricter than writing code yourself. Your code undergoes review before merging, but if you approve a PR, it could be merged directly into our main branch, leaving no room for errors.

**Review by Wendong-Fan — APPROVED** [review]

**Review by harryeqs — APPROVED** [review]

**Pull request merged**